kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/util/NetUtil.java

```diff
@@ -26,27 +27,45 @@ public static String hardwareAddressToString(byte[] macAddress) {
         if (macAddress == null) {
             return "N/A";
         }
-        StringBuilder sb = new StringBuilder();
-        for (int i = 0; i < macAddress.length; i++) {
-            sb.append(String.format("%02X%s", macAddress[i], i < macAddress.length - 1 ? ":" : ""));
+
+        if (macAddress.length != 6) {
+            throw new IllegalArgumentException("macAddress is invalid");
```

Comment:
As a reminder, the validation is correct but it will throw an runtime exception that was not previously thrown and that the caller will likely not catch.
I guess we will find problems, if any, during release QA